src/ensembl/src/content/app/browser/browser-nav/BrowserNavBar.tsx

```diff
@@ -1,50 +1,99 @@
-import React, { FunctionComponent } from 'react';
+import React, { useEffect } from 'react';
 import { connect } from 'react-redux';
 import classNames from 'classnames';
 
-import { browserNavConfig, BrowserNavItem } from '../browserConfig';
+import BrowserNavIcon from './BrowserNavIcon';
+import BrowserRegionEditor from '../browser-region-editor/BrowserRegionEditor';
+import BrowserRegionField from '../browser-region-field/BrowserRegionField';
 
 import { RootState } from 'src/store';
-import { getBrowserNavStates } from '../browserSelectors';
+import { browserNavConfig, BrowserNavItem } from '../browserConfig';
+import {
+  getBrowserNavStates,
+  getChrLocation,
+  getRegionEditorActive,
+  getRegionFieldActive
+} from '../browserSelectors';
+import {
+  toggleRegionEditorActive,
+  toggleRegionFieldActive
+} from '../browserActions';
 import { getIsTrackPanelOpened } from '../track-panel/trackPanelSelectors';
-import { BrowserNavStates } from '../browserState';
-
-import BrowserNavIcon from './BrowserNavIcon';
+import { BrowserNavStates, ChrLocation } from '../browserState';
+import { getGenomeKaryotypes } from 'src/genome/genomeSelectors';
+import { GenomeKaryotype } from 'src/genome/genomeTypes';
```

Comment:
The term "karyotype" describes the complete set of chromosomes of an organism. Therefore, a genome has one karyotype (not "karyotypes", as in the name of the `getGenomeKaryotypes ` selector); and the type that is called `GenomeKaryotype` in the line above is in fact a `GenomeKaryotypeItem`.